### DISCUSSION TITLE
Getting started

### DISCUSSION THREAD

**Discussion by etaloncop** [schismtracker/schismtracker#441] [Q&A] [answered]
Heya, kinda new in this, 
You guys dont really have any getting started guide of some sort, do you? and if so why its not linked in the readme?
I would very much appreciate at least some basic sample library (maybe sine pulse sawtooth etc.) to get started. It is really confusing for a person that finds this repo and wants to give it a shot. As I download it and I dont have anything to get started making music, right?
a) Where do I get some basic samples
b) How do I do looping, it doesnt work for me (and the F1 help doesnt really help either)
![image](https://github.com/schismtracker/schismtracker/assets/136369289/0f176b9d-d6f1-41fe-a553-5968e5cee694)
c) Instruments, what are they? Why dont i just use samples?
d) can I add multiple samples at once? maybe whole directory?
e) can I loop just part of the pattern?
f) Are there any presets to learn from? What are the keywords i would search for? For some reason just searching with schism keyword doesnt yield many results.

**Comment by mrpapersonic** [comment, marked as answer]
> a) Where do I get some basic samples

You can get many samples from [existing](https://modarchive.org) [modules](http://modland.com/pub/modules/Impulsetracker/). In fact, most modules already use samples ripped from other modules. Schism directly supports importing samples from modules, too.
> b) How do I do looping, it doesnt work for me (and the F1 help doesnt really help either)

For samples to loop, you have to set the loop method to either On Forwards or On Ping Pong, and then set the loop begin and end points. The sustain loop is largely irrelevant until instrument mode is enabled.
> c) Instruments, what are they? Why dont i just use samples?

Instruments let you add volume, pan, and pitch/filter effects on a sample and are useful for e.g. chiptunes. You can also set different samples to be played at different pitches, which is useful for making nice sounding instruments like a piano.
> d) can I add multiple samples at once? maybe whole directory?

I don't think so, you have to load them all in separately.
> e) can I loop just part of the pattern?

Yep. You can set the loop start with SB0 and mark the end with SBx, with x being the amount of times you want it to loop.
> f) Are there any presets to learn from? What are the keywords i would search for? For some reason just searching with schism keyword doesnt yield many results.

Schism is a modern clone(?) of Impulse Tracker for DOS. You can find a lot of [material for using IT](https://ia802805.us.archive.org/25/items/ImpulseTrackerIIUserManual/Impulse%20Tracker%20II%20User%20Manual.pdf) that can be applied to Schism as well.

**Comment by etaloncop** [comment]
Thank you so much for such a quick reply, I cannot thank you enough, you are awesome

**Comment by etaloncop** [comment]
Damn, finally figured out how to do portamento, thats sick

**Comment by etaloncop** [comment]
OOoh this is so much fun

**Comment by etaloncop** [comment]
I need a way to apply more effects

**Comment by mrpapersonic** [comment]
You can't apply more than one effect at a time, but there are some effects that have equivalents in the volume column (portamento up, down, and tone portemento are notable examples). If you truly do need more effects, [Renoise](https://www.renoise.com) has that ability.

**Comment by faithanalog** [comment]
there's also an old manual from 2007 at https://archive.org/details/SchismTrackerUsersManual

it's quite useful and most of it still applies since schism has not had a lot of churn as a program, though it is wrong about some limitations schism doesn't have anymore / unaware of features that were added after that point. I was referencing it to learn how to set up the advanced midi configuration

I also highly recommend exploring other peoples' files to see what they do. It's great because you can hear it, play it, watch things happen, make changes and see what it did. This is what really got me into using schism and out of the "blank canvas and i dont know what anything does" pit. http://modarchive.org/ has a lot!

**Comment by JesusGod-Pope666-Info** [comment]
How do you start the app in Linux? some kind of run code in the terminal?

**Reply by mrpapersonic** [reply]
You need to run schism with the LD_LIBRARY_PATH environment variable set to the path with the `schismtracker` and related `*.so` files, or copy said `*.so` files to /usr/local/lib.

**Reply by JesusGod-Pope666-Info** [reply]
.... Is it so hard to just make things portable and run out of the box.

Sick and tired of Linux - it is crap.

**Reply by JesusGod-Pope666-Info** [reply]
Seems it is working on Windows 7 out of the box, as it should. so.... I guess I will try that.

**Reply by mrpapersonic** [reply]
When you're on linux you should really just be using your package manager anyway instead of downloading binaries like on windows. It's a completely different operating system. It's not "crap". You're just not used to it ;)

For example on ubuntu/debian/mint you would use `sudo apt-get install schism`. Alas you can obviously keep using windows but I'd advise against using windows 7 mostly because of the security holes waiting to be penetrated and also that lots of software are just dropping support for it even when its trivial to keep it working.

**Reply by JesusGod-Pope666-Info** [reply]
> When you're on linux you should really just be using your package manager anyway instead of downloading binaries like on windows. It's a completely different operating system. It's not "crap". You're just not used to it ;)
> 
> For example on ubuntu/debian/mint you would use `sudo apt-get install schism`. Alas you can obviously keep using windows but I'd advise against using windows 7 mostly because of the security holes waiting to be penetrated and also that lots of software are just dropping support for it even when its trivial to keep it working.

Well I still think it is crap, the system is based on the wrong foundation and that will not change - everything took a nose dive after windows 2000. And Linux just needs to much nonsense to get things working, code and all.

Anyway, anyone knows how to get the "keyboard" on Schism Tracker? I can't seem to find it. I think I sorta got the F's keys sorted.

Having to install first with code, then guess the name of the running file, like install schism yet the file to run it is schismtracker - like come on. Everything should be portable out of the package.

**Reply by JesusGod-Pope666-Info** [reply]
![Screenshot_2024-12-24_02-46-03](https://github.com/user-attachments/assets/67206497-6f5f-432d-84b1-8cc6e19b7597)

**Reply by JesusGod-Pope666-Info** [reply]
And now we have all this political nonsense mixed with software ruining things even further. It is madness. Anyway, thanks for the tip installing it on Linux. Got it working on both Windows and Linux now - but..... I can't seem to find this keyboard thing, or figure out what is going on with Boo's lullaby, but I guess I don't know much in regards of trackers.
It is one of the best tunes I have found for Trackers.

**Reply by JesusGod-Pope666-Info** [reply]
@mrpapersonic did we not have a thread with help on things somewhere? I can't seem to find it, I think you ticked it as finish and responded. As I forgotten some of the shortcuts...... Typically.
